src/Authentication/Actions/GetHomepageUrlAction.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Arpite\Authentication\Actions;
+
+use Arpite\Authentication\Exceptions\HomepageNotFoundException;
+
+class GetHomepageUrlAction
+{
+	/**
+	 * @throws HomepageNotFoundException
+	 */
+	public function execute(): string
+	{
+		$routeServiceProviderClass = "App\Providers\RouteServiceProvider";
```

Comment:
You could declare type
`/** @var class-string $routeServiceProviderClass */`
then I think phpstan would not complain when using function `method_exists`